selendroid-server/src/main/java/io/selendroid/server/model/DefaultAppiumUiAutomatorDriver.java

```diff
@@ -0,0 +1,279 @@
+
+    public static final String BROWSER_NAME = "browserName";
+    public static final String PLATFORM = "platform";
+    public static final String SUPPORTS_JAVASCRIPT = "javascriptEnabled";
+    public static final String TAKES_SCREENSHOT = "takesScreenshot";
+    public static final String SUPPORTS_ALERTS = "handlesAlerts";
+    public static final String ROTATABLE = "rotatable";
+    public static final String ACCEPT_SSL_CERTS = "acceptSslCerts";
+    public static final String SUPPORTS_NETWORK_CONNECTION = "networkConnectionEnabled";
+    public static final String AUTOMATION_NAME = "automationName";
+    private ServerInstrumentation serverInstrumentation = null;
+    private KeySender keySender = null;
+    private Session session = null;
+
+    public DefaultAppiumUiAutomatorDriver(ServerInstrumentation serverInstrumentation) {
+        this.keySender = new InstrumentedKeySender(serverInstrumentation);
+        this.serverInstrumentation = serverInstrumentation;
+    }
+
+    @Override
+    public AndroidElement findElement(By by) {
+        throw new NoSuchMethodImplementationException("Method not implemented");
+    }
+
+    @Override
+    public List<AndroidElement> findElements(By by) {
+        throw new NoSuchMethodImplementationException("Method not implemented");
+    }
+
+    @Override
+    public String getCurrentUrl() {
+        throw new NoSuchMethodImplementationException("Method not implemented");
+    }
+
+    @Override
+    public Session getSession() {
+        return session;
+    }
+
+    @Override
+    public JSONObject getSessionCapabilities(String sessionId) {
+        SelendroidLogger.info("session: " + sessionId);
+
+        JSONObject copy;
+        try {
+            JSONObject capabilities = session.getCapabilities();
+            if (capabilities != null) {
+                copy = new JSONObject(capabilities.toString());
+            } else {
+                copy = new JSONObject();
+            }
+            copy.put(TAKES_SCREENSHOT, true);
+            copy.put(BROWSER_NAME, "selendroid");
+            copy.put(AUTOMATION_NAME, "uiautomator");
+            copy.put("platformName", "android");
+            copy.put("platformVersion", serverInstrumentation.getOsVersion());
+            copy.put(ROTATABLE, true);
+            copy.put(PLATFORM, "android");
+            copy.put(SUPPORTS_ALERTS, true);
+            copy.put(SUPPORTS_JAVASCRIPT, true);
+            copy.put(SUPPORTS_NETWORK_CONNECTION, true);
+            copy.put("version", serverInstrumentation.getServerVersion());
+            copy.put(ACCEPT_SSL_CERTS, true);
+            SelendroidLogger.info("capabilities: " + copy);
+            return copy;
+        } catch (JSONException e) {
+            throw new SelendroidException(e);
+        }
+    }
+
+    @Override
+    public JSONObject getFullWindowTree() {
+        throw new NoSuchMethodImplementationException("Method not implemented");
```

Comment:
Rather than repeat the string every time, the exception should just default to using `"Method not implemented"` if the message isn't set explicitly.